File: 1431-Kids-With-the Greatest Number of Candies/solution.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        ArrayList<Boolean> arr = new ArrayList<>();
        int max = 0;
        for (int i = 0; i <candies.length; i++) {
            if(max <= candies[i]){
                max = candies[i];
            }
        }
        for (int i = 0; i <candies.length; i++) {
            int sum = candies[i] + extraCandies;
            if(sum >= max){
                arr.add(true);
            }else {
                arr.add(false);
            }
        }
        return arr;
    }
}

File: 1431-Kids-With-the Greatest Number of Candies/solution.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max_candies = max(candies)
        result = []
        for i in range(len(candies)):
            if candies[i] + extraCandies >= max_candies:
                result.append(True)
            else:
                result.append(False)
        return result